docs/RFCS/20230605_workload_level_index_recommendation.md

```diff
@@ -0,0 +1,348 @@
+borrow the idea from [Huffman coding](https://en.wikipedia.org/wiki/Huffman_coding), each time we merge the most 
+expansive index with its nearest indexes (may from the leaf node or internal node) and then repeat this process until 
+the budget satisfied. Just like the example above, we merge the indexes `(k, i)` and `(k, j)` to `(k)`. If there is no 
+sharing between existing indexes, we can get rid of those expansive one. 
+
+In order to complete the algorithm above, we need to think of some objective function to estimate the cost for different 
+indexes based on the number of indexed columns, the number of storing columns, different types of columns, usage in the 
+whole workload, etc. 
+
+As mentioned in the finding index candidates, the frequency is an important aspect we should pay attention to, in 
+`system.statement_statistics`, there is a column called `statistics` containing the times of on a type of query has 
+been executed, we can use this to assign a weight for each node in the tree so that the merging algorithm will cover the 
+importance of the indexes (by frequency). If we collect from the raw sql statements mentioned in the future work, we can 
+re-compute the frequency from scratch and use such more precise information to assign the weight.
+
+## Drop index
+
+As for the `DROP INDEX`, since we select all the leaf nodes before, so we can directly drop them without hesitation 
+(covered by all the leaf nodes). Nevertheless, we only select some internal nodes to build indexes  when we need to 
+merge some leaf nodes for cost optimization so that not everything in the tree is covered, meaning that some dropped 
+indexes are not covered by the indexes selected to create. Since the space and time budget is confined, we can also take 
+dropping indexes into account, the cost function can be used to calculate the space released by dropping one index. 
+Following the idea of the trie tree, we can add all the indexes to drop to the tree. When we decide to merge two 
+indexes, there may exist some indexes to drop along the paths to their LCA, when we compute the space to save, we can 
+minus those space occupied by the indexes to drop since we can't cover them once we merge these two indexes. Thus, our 
+merging process will be more comprehensive.
+
+## Rationale and alternatives
+
+...
+
+# Unresolved questions
+
+## References
+
+### KV layer representations for indexes
+Generally, key representation for the index is `/tableID/indexID/indexColumns.../primaryKey`, the value will be 
+`[/StoreColumns]`. Suppose there are 3 tuples inserted to the table `t`: `{1, 2, 3}, {4, 5, 6}, {7, 8, 9}` and i is the 
+primary key. If we create the following index:
+```sql
+CREATE INDEX ON t(j, k)
+```
+Assume the tableID is 1 and the indexID is also 1. The key-value pairs for this index will be:
+
+| Key        | Value | 
+|------------|-------|
+| /1/1/2/3/1 | Ø     |
+| /1/1/5/6/4 | Ø     |
+| /1/1/8/9/7 | Ø     | 
+As for an index with storing:
```

Comment:
nit: put this on the next line since it looks like it's part of the table right now